spring-data-microservice-transaction-sample/README.md

```diff
@@ -0,0 +1,550 @@
+These repositories are available only to users with a commercial license and permission.
+To get a license and permission, please [contact us](https://scalar-labs.com/contact_us/).
+
+You also need the `gpr.user` property for your GitHub username and the `gpr.key` property for your personal access token.
+You must either add these properties in `~/.gradle/gradle.properties` or specify the properties by using the `-P` option when running the `./gradlew` command as follows:
+
+```shell
+$ ./gradlew run ... -Pgpr.user=<YOUR_GITHUB_USERNAME> -Pgpr.key=<YOUR_PERSONAL_ACCESS_TOKEN>
+````
+
+Or you can also use environment variables, `USERNAME` for your GitHub username and `TOKEN` for your personal access token.
+
+```shell
+$ export USERNAME=<YOUR_GITHUB_USERNAME>
+$ export TOKEN=<YOUR_PERSONAL_ACCESS_TOKEN>
+```
+
+For more details, please see [Install - ScalarDB SQL](https://github.com/scalar-labs/scalardb-sql#install).
+
+## Sample application
+
+### Overview
+
+This tutorial describes how to create a Spring Boot sample application for Microservice Transaction for the same use case as [ScalarDB Sample](https://github.com/scalar-labs/scalardb-samples/tree/main/scalardb-sample) but by using Two-phase Commit Transactions in ScalarDB.
+
+There are two microservices called *Customer Service* and *Order Service* based on the [*Database-per-service* pattern](https://microservices.io/patterns/data/database-per-service.html) in this sample application.
+
+Customer Service manages customers' information including credit card information like a credit limit and a credit total.
+Order Service is responsible for order operations like placing an order and getting order histories.
+Each service has gRPC endpoints. Clients call the endpoints, and the services call the endpoints each other as well.
+Customer Service and Order Service uses MySQL and Cassandra through ScalarDB, respectively.
+
+![Overview](images/overview.png)
+
+Note that both services access a small coordinator database used for the Consensus Commit protocol.
+The database is service-independent and exists for managing transaction metadata for Consensus Commit in a highly available manner, so we don't think it is violating the database-per-service pattern.
+*NOTE: We also plan to create a microservice container for the coordinator database to truly achieve the database-per-service pattern.*
+
+In this sample application, for ease of setup and explanation, we co-locate the coordinator database in the same Cassandra instance of the Order Service, but of course, the coordinator database can be managed as a separate database.
+
+Also, note that application-specific error handling, authentication processing, etc., are omitted in the sample application since it focuses on explaining how to use ScalarDB.
+Please see [this document](https://github.com/scalar-labs/scalardb/blob/master/docs/api-guide.md#handle-exceptions) for the details of how to handle exceptions in ScalarDB.
```

Comment:
```suggestion
Please see [this document](https://github.com/scalar-labs/scalardb/blob/master/docs/two-phase-commit-transactions.md#handle-exceptions) for the details of how to handle exceptions in ScalarDB.
```
Should we refer to the 2pc exception guide instead ?

Reply:
Oh, definitely yes in this sample app. Thank you!